python/ray/tune/utils/util.py

```diff
@@ -504,8 +502,8 @@ def tune_func(config):
             raise RuntimeError(f"No GPU ids found from {ray.get_gpu_ids()}. "
                                "Did you set Tune resources correctly?")
         gpu_id = gpu_id_list[0]
-    gpu_object = GPUtil.getGPUs()[gpu_id]
     for i in range(int(retry)):
+        gpu_object = GPUtil.getGPUs()[gpu_id]
```

Comment:
`gpu_id` is a string, but `getGPUs()` returns a list.  You probably need to compare against the GPU id (which is also an int) returned in the list.